Add Custom Translation Navigation Screen

JoshuaSchaefer opened the pull request:
Add a navigation screen to allow users to navigate between different defined translations. Also, separate adding and editing a custom translation into two different processes.

Note: Tests are passing, they are just being flaky
---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [ ] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [ ] Choose the proper base branch (*develop*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no warnings and errors 
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [ ] Perform a self-review of the changes
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [ ] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [ ] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Additional Description
The new navigation screen displays buttons that link to each defined translation. Then a separate button to add a new custom translation. When the user adds a new translation, they can select which language they want to define and cancel or save any additions. When a user edits an existing translation, they cannot change the language, but they can delete the translation and cancel or save any changes.

Also, with this change, most of the functionality in the following document is implemented.
[catrobat custom translations](https://docs.google.com/document/d/1IU-12Np7yajbz2Gt9rkfm2jasQlt7trhliu1gqE1k4Y/edit?usp=sharing)

### Tests - additional information
All tests that are impacted by this change have been updated.

danielzhang130 commented on assets/js/components/ProjectEditor.js:
```diff
@@ -103,15 +120,17 @@ export function ProjectEditor (projectDescriptionCredits, programId, textFields,
   this.close = () => {
     $(window).off('popstate', this.popStateHandler)
-    showDefaultTopBarTitle()
+    $(window).on('popstate', this.navigationHandler)
```
Should this be in reopenNavigation or does it break things?

JoshuaSchaefer replied:
Moved to reopenNavigation

danielzhang130 commented on assets/js/components/ProjectEditor.js:
```diff
@@ -103,15 +120,17 @@ export function ProjectEditor (projectDescriptionCredits, programId, textFields,
 
-    this.body.removeClass('overflow-hidden')
     this.editTextUI.addClass('d-none')
+    this.editTextNavigation.removeClass('d-none')
```
same here, Should this be in reopenNavigation or does it break things?

JoshuaSchaefer replied:
Moved to reopenNavigation

danielzhang130 commented on assets/js/components/ProjectEditor.js:
```diff
@@ -65,7 +65,17 @@ export function ProjectEditor (projectDescriptionCredits, programId, textFields,
-  this.show = () => {
+  this.show = (navigationHandler, navigationCallback, language, showLanguageSelect, showDeleteButton, headerText) => {
+    this.navigationHandler = navigationHandler
+    this.navigationCallback = navigationCallback
```
I find the handler and callback confusing. see the other comments

JoshuaSchaefer replied:
Removed the navigation handler. I added the navigation callback later and didn't realize that the navigation handler and some other code could be moved in there. Should be fixed now

danielzhang130 reviewed (commented):
overall, i think some of the logic of closing ProjectEditor might belong to ProjectEditorNavigation

danielzhang130 reviewed (commented):
is there a way to disable the left navigation panel?

JoshuaSchaefer commented:
> 

There might be, I haven't seen any code that would do it, but I can look into it if we want to do that. However, I don't think it causes any problems with the design if it is enabled though, right?

danielzhang130 commented:
it's not working for me. When i open an existing translation, it just shows the default text

danielzhang130 commented:
> > 
> 
> There might be, I haven't seen any code that would do it, but I can look into it if we want to do that. However, I don't think it causes any problems with the design if it is enabled though, right?

i kept opening the drawer when i try to hit the back button. maybe it's just me

JoshuaSchaefer commented:
> > > 
> > 
> > 
> > There might be, I haven't seen any code that would do it, but I can look into it if we want to do that. However, I don't think it causes any problems with the design if it is enabled though, right?
> 
> i kept opening the drawer when i try to hit the back button. maybe it's just me

I haven't had any problems with it. I could see it being sensitive, but I think it should be okay for now. We can disable it in the future if we are having trouble with it

JoshuaSchaefer commented:
> it's not working for me. When i open an existing translation, it just shows the default text

Sorry, I was calling the wrong method. Should be fixed now 👍

danielzhang130 reviewed (commented):
Will there be new tests for this navigation screen?

JoshuaSchaefer commented:
> Will there be new tests for this navigation screen?

I think the modifications/updates made to the exiting tests practically test all the new functionality through navigating to different screens then adding and editing translations. So, I think new tests for this screen might be a little redundant.

danielzhang130 reviewed (approved):
OK. as long as the failing tests actually pass locally

Commit:
Add Custom Translation Navigation Screen

Add a navigation screen to allow users to navigate between different defined translations. Also, separate adding and editing a custom translation into two different processes.
